tensorizer/serialization.py

```diff
@@ -481,46 +509,46 @@ def __init__(
         )
 
         # Placeholders
-        if include_crc32:
+        if self.include_crc32:
             self.add_crc32(0)
-        if include_sha256:
+        if self.include_sha256:
             self.add_sha256(b"")
-        if crypt_info is not None:
-            crypt_info.sized_pack_into(self.buffer, self.crypt_info_offset)
+        if self.crypt_info is not None:
+            self.crypt_info.sized_pack_into(self.buffer, self.crypt_info_offset)
 
         self.data_length_segment.pack_into(
-            self.buffer, self.data_length_offset, data_length
+            self.buffer, self.data_length_offset, self.data_length
         )
 
-        metadata_entry_segment: struct.Struct = struct.Struct(
-            self.metadata_entry_segment_template.format(
-                name_len=name_len,
-                dtype_len=dtype_len,
-                shape_len=shape_len,
-            )
-        )
+        metadata_entry_segment = self.get_metadata_entry_segment()
 
         self.metadata_entry = metadata_entry_segment.pack(
-            name_len,  # Name length
-            name,  # Name
-            tensor_type.value,  # Whether this is a parameter or a buffer
-            dtype_len,  # Dtype length
-            dtype,  # Dtype
-            shape_len,  # Shape length
-            *shape,  # Shape
-            file_offset,  # Header start (relative to the file)
+            self.name_len,  # Name length
+            self.name,  # Name
+            self.tensor_type.value,  # Whether this is a parameter or a buffer
+            self.dtype_len,  # Dtype length
+            self.dtype,  # Dtype
+            self.shape_len,  # Shape length
+            *self.shape,  # Shape
+            self.file_offset,  # Header start (relative to the file)
             # Tensor data start (relative to the file):
-            file_offset + self.data_offset,
-            data_length,  # Tensor length
+            self.data_offset,
+            self.data_length,  # Tensor length
+        )
```

Comment:
imo metadata is a dead-end. Keeping it intact for now makes it easier to load 2.x tensors, but in future major versions everything would be in header. Organizationally they are already the same as headers in 3.x

Reply:
That's fair. The fixed format for metadata has been problematic before because e.g. `data_length` would be far more useful if split into `serialized_length` and `deserialized_length` to separately communicate the file layout and the amount of RAM/VRAM usage the deserializer will need to anticipate. If it is only used as a shortcut to navigate the more detailed header section that follows it, then I'd be equally fine with it being a mostly-dead section (or just removing it now from 3.0 files and keeping only its deserializer).